src/Slices/Status/UI/SupportArchive.tsx

```diff
@@ -4,30 +4,47 @@ import {
   AlertActionCloseButton,
   AlertGroup,
 } from "@patternfly/react-core";
-import { Either } from "@/Core";
+import { useGetSupportArchive } from "@/Data/Managers/V2/Support/GetSupportArchive";
 import { DependencyContext } from "@/UI/Dependency";
 import { DownloadButton, Phase } from "./Components";
 
+/**
+ * SupportArchive component is responsible for managing the download
+ * of the support archive. It utilizes the useGetSupportArchive hook
+ * to handle the download process and provides user feedback through
+ * a DownloadButton and alerts for any errors that may occur during
+ * the download.
+ *
+ * @returns {JSX.Element} The rendered component.
+ */
 export const SupportArchive: React.FC = () => {
-  const { commandResolver, archiveHelper } = useContext(DependencyContext);
+  const { archiveHelper } = useContext(DependencyContext);
   const [phase, setPhase] = useState<Phase>("Default");
```

Comment:
Can't we use the isPending fro mthe hook to know if we need to display a spinner instead?

Reply:
I didn't want to change more than is essential, I would have to change buttons, and I felt like it's good enough